scapy/fields.py

```diff
@@ -900,6 +900,190 @@ def any2i(self, pkt, x):
     def i2repr(self, pkt, x):
         return _EnumField.i2repr(self, pkt, x)
 
+
+class LEBitField(BitField):
+    """
+    a little endian version of the BitField
+    """
+
+    def addfield(self, pkt, s, val):
+        """
+
+        :param pkt: packet instance the raw string s and field belongs to
+        :param s:   raw string representing the frame
+        :param val: value
+        :return: final raw string, tuple (s, bitsdone, data) if in between bit field
+
+        as we don't know the final size of the full bitfield we need to accumulate the data.
+        if we reach a field that ends at a octet boundary, we build the whole string
+
+        !!! NEVER MIX LEBitField with BitField !!!
```

Comment:
I think so and I will add an exception. It was not completely clear to me how to handle that kind of internal error as I was not able to derive some kind of rule by locking at the remaining code base :smile: